### DISCUSSION TITLE
Libreview Account ID retrieve failing - HTTP 504 Error

### DISCUSSION THREAD

**Discussion by rpm5099** [maheini/FreeStyle-Libre-3-patch#36] [Q&A]
The URL that the current version of Juggluco apk attempts to access to retrieve the Libreview Account ID is apparently no longer valid.  I was able to see a partial HTTP 504 error in the text area where the account number would be.  I'm able to help/contribute if desired.

**Comment by rpm5099** [comment]
The fact that this is happening makes me realize that this is part of the playbook they use, where Abbott is actively thwarting usage of 3rd party apps, and in addition the Libreview site seems to require 2 factor authentication.  Perhaps a fallback script to retrieve the account id, assuming that it's still exposed via the web api, would be helpful?  I have not looked through source, just trying it out for the first time here.

**Comment by maheini** [comment]
Hi @rpm5099 
I don't think it will be restricted, because Juggluco is using the native FSTL3 library and therefore the Libre 3 app shouldn't work as well. Thank you for submitting your note. Have you tried switching back to version 4.2.7 ? [Download link here](https://github.com/maheini/FreeStyle-Libre-3-patch/raw/main/Juggluco-solution/versions/4.2.7/Juggluco.apk).

Otherwhise, I guess it might be a bug or problem within Juggluco. It would be very interesting to know if Juggluco 4.2.7 has the same behaviour, as this version is proofed working (and very stable)

**Comment by rpm5099** [comment]
Are you certain that Juggluco is using the native FSL3 library for that specific function - i.e. retrieving the users Libreview account number?  It's part of the setup in step 3 of your readme where it says `In the next screen you have to enter your login data for Libreview. It must be the account with which the sensor was activated. Then click on "Get Account ID".`  [Here](https://github.com/maheini/FreeStyle-Libre-3-patch/blob/main/Juggluco-solution/juggluco-direct-instructions/en/instructions.md)

**Reply by maheini** [reply]
Well, to be honest I'm not sure about the Libreview function. At least the decryption is using the native library. But it might probably be like so. I'm not saying it's using Libre 3 app for libreview communication - only the library, which is communicating in the background is integrated within Juggluco.

**Comment by rpm5099** [comment]
Also - are you talking about this library?  I can't find anything else that it would be - https://fsl.fmrib.ox.ac.uk/

**Reply by maheini** [reply]
It's not a public library of any kind - it's extracted from within the original Libre 3 apk, done by jkaltes. However, not confirmed and without any guarantee.

**Comment by rpm5099** [comment]
Oh ok, understood.  I will check on that other version at some point.  Thanks for your work on this, the Libre 3 app is really pretty bad.
